src/TaskService.Validation/Task/EditTaskRequestValidator.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using System.Collections.Generic;
+using FluentValidation;
+using FluentValidation.Validators;
+using LT.DigitalOffice.Kernel.Validators;
+using LT.DigitalOffice.TaskService.Data.Interfaces;
+using LT.DigitalOffice.TaskService.Models.Dto.Enums;
+using LT.DigitalOffice.TaskService.Models.Dto.Requests;
+using LT.DigitalOffice.TaskService.Validation.Task.Interfaces;
+using Microsoft.AspNetCore.JsonPatch.Operations;
+
+namespace LT.DigitalOffice.TaskService.Validation.Task
+{
+  public class EditTaskRequestValidator : BaseEditRequestValidator<EditTaskRequest>, IEditTaskRequestValidator
+  {
+    private readonly ITaskPropertyRepository _taskPropertyRepository;
+
+    private void HandleInternalPropertyValidation(Operation<EditTaskRequest> requestedOperation, CustomContext context)
+    {
+
+      #region paths
+
+      AddСorrectPaths(
+        new List<string>
+        {
+          nameof(EditTaskRequest.Name),
+          nameof(EditTaskRequest.Description),
+          nameof(EditTaskRequest.AssignedTo),
+          nameof(EditTaskRequest.PriorityId),
+          nameof(EditTaskRequest.StatusId),
+          nameof(EditTaskRequest.TypeId),
+          nameof(EditTaskRequest.PlannedMinutes)
+        });
+
+      AddСorrectOperations(nameof(EditTaskRequest.Name), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditTaskRequest.Description), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditTaskRequest.AssignedTo), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditTaskRequest.PriorityId), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditTaskRequest.StatusId), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditTaskRequest.TypeId), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditTaskRequest.PlannedMinutes), new List<OperationType> { OperationType.Replace });
+
+      #endregion
+
+      #region firstname
+
+      AddFailureForPropertyIf(
+        nameof(EditTaskRequest.Name),
+        x => x == OperationType.Replace || x == OperationType.Add,
+        new()
+        {
+          { x => !string.IsNullOrEmpty(x.value.ToString()), "Name is empty." },
+          { x => x.value.ToString().Length < 150, "Name is too long" }
+        });
+
+      #endregion
+
+      #region description
+
+      AddFailureForPropertyIf(
+        nameof(EditTaskRequest.Description),
+        x => x == OperationType.Replace || x == OperationType.Add,
+        new()
+        {
+          {
```

Comment:
i think better x => x.value?.ToString().Trim().Length < 150, "Description is too long"

Reply:
fixed